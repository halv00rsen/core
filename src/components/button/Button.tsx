import React, { Ref } from 'react';
import styled from 'styled-components';
import { ButtonColor, getButtonColor } from '../../ui/color';

interface ButtonUIProps {
  backgroundColor: string;
}

const ButtonUI = styled.button<ButtonUIProps>`
  border: none;
  border-radius: 5px;
  padding: 7px 9px;
  text-decoration: none;
  cursor: pointer;
  background-color: ${(props) => props.backgroundColor};
  opacity: 1;
  transition: 0.3s;
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: 700;
  color: #ebebeb;

  &:hover {
    opacity: 0.9;
  }
`;

export interface ButtonProps {
  backgroundColor?: ButtonColor;
}

export const Button = ({
  children,
  backgroundColor = 'blue',
  ...rest
}: ButtonProps &
  JSX.IntrinsicElements['button'] & { ref?: Ref<HTMLButtonElement> }) => {
  return (
    <ButtonUI backgroundColor={getButtonColor(backgroundColor)} {...rest}>
      {children}
    </ButtonUI>
  );
};
