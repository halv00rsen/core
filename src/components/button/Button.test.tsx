import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './Button';

it('renders correctly', () => {
  const tree = renderer.create(<Button>Main button</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('accepts color', () => {
  const tree = renderer
    .create(<Button backgroundColor={'green'}>Click me</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with onClick function', () => {
  const tree = renderer
    .create(<Button onClick={console.log}>Click</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
