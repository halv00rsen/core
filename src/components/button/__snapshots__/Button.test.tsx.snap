// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts color 1`] = `
<button
  className="Button__ButtonUI-sc-1b6loj5-0 ibYuGU"
>
  Click me
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="Button__ButtonUI-sc-1b6loj5-0 eFxyfD"
>
  Main button
</button>
`;

exports[`renders with onClick function 1`] = `
<button
  className="Button__ButtonUI-sc-1b6loj5-0 eFxyfD"
  onClick={[Function]}
>
  Click
</button>
`;
